tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -44,10 +45,33 @@
 EXAMPLES_KEY = 'examples'
 
 
-def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int]) -> int:
+def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int], feature_name: Text) -> int:
   assert num_partitions == len(
       buckets), 'Partitions do not match bucket number.'
-  bucket = int(hashlib.sha256(record).hexdigest(), 16) % buckets[-1]
+
+  if feature_name:
+    # Use given feature for partitioning the examples.
+    # Deserialize the record to tf.train.Example. 
+    example = tf.train.Example()
+    example.ParseFromString(record)
+    feature_list = example.features.feature[feature_name]
+    
+    # Initialize the feature value.
+    feature_value = ""
+    if feature_list.bytes_list.value:
+      feature_value = feature_list.bytes_list.value[0]
+    elif feature_list.float_list.value:
+      feature_value = feature_list.float_list.value[0]
+    elif feature_list.int64_list.value:
+      feature_value = feature_list.int64_list.value[0]
+    
+    # Convert the feature value to bytes for the fingerprint.
+    bytes_value = bytes(str(feature_value), "ascii")
+    bucket = int(hashlib.sha256(bytes_value).hexdigest(), 16) % buckets[-1]
+  else:
+    # Use the entire record.
+    bucket = int(hashlib.sha256(record).hexdigest(), 16) % buckets[-1]
```

Comment:
You don't need to duplicate the line 70 & 73, you can do sth like this:

partition_str = record
if feature:
....partition_str = xxx
bucket = int(hashlib.sha256(partition_str).hexdigest(), 16) % buckets[-1]